src/typechecker.erl

```diff
@@ -1951,7 +1951,7 @@ do_type_check_expr_in(Env, Ty, Char = {char, LINE, _}) ->
         {true, Cs} ->
            {#{}, Cs};
        false ->
-           throw({type_error, char_expr, LINE, Char, Ty})
+           throw({type_error, Char, {type, erl_anno:new(0), char, []}, Ty})
```

Comment:
nitpicking: could use `type(char)` (and `type(list)` below) for brevity and consistency.
Also `LINE` in the function head seems to be unused

Reply:
Absolutely, thanks!